ladybug_display_schema/display3d.py

```diff
@@ -14,7 +18,196 @@ class DisplayVector3D(DisplayBaseModel):
+
+    geometry: Point3D = Field(
+        ...,
+        description='Point3D for the geometry.'
+    )
+
+
+class DisplayRay3D(SingleColorBase):
+    """A ray object in 3D space with display properties."""
+
+    type: constr(regex='^DisplayRay3D$') = 'DisplayRay3D'
+
+    geometry: Ray3D = Field(
+        ...,
+        description='Ray3D for the geometry.'
+    )
+
+
+class DisplayPlane(SingleColorBase):
+    """A plane object with display properties."""
+
+    type: constr(regex='^DisplayPlane$') = 'DisplayPlane'
+
+    geometry: Plane = Field(
+        ...,
+        description='Plane for the geometry.'
+    )
+
+    show_axes: bool = Field(
+        False,
+        description='A boolean to note whether the plane should be displayed '
+        'with XY axes instead of just an origin point and a normal vector.'
+    )
+
+    show_grid: bool = Field(
+        False,
+        description='A boolean to note whether the plane should be displayed '
+        'with a grid.'
+    )
+
+
+class DisplayLineSegment3D(LineCurveBase):
+    """A single line segment face in 3D space with display properties."""
+
+    type: constr(regex='^DisplayLineSegment3D$') = 'DisplayLineSegment3D'
+
+    geometry: LineSegment3D = Field(
+        ...,
+        description='LineSegment3D for the geometry.'
+    )
+
+
+class DisplayPolyline3D(LineCurveBase):
+    """A polyline in 3D space with display properties."""
+
+    type: constr(regex='^DisplayPolyline3D$') = 'DisplayPolyline3D'
+
+    geometry: Polyline3D = Field(
         ...,
-        description='Color for the geometry.'
+        description='Polyline3D for the geometry.'
+    )
+
+
+class DisplayArc3D(LineCurveBase):
+    """A single arc or circle in 3D space with display properties."""
+
+    type: constr(regex='^DisplayArc3D$') = 'DisplayArc3D'
+
+    geometry: Arc3D = Field(
+        ...,
+        description='Arc3D for the geometry.'
+    )
+
+
+class DisplayFace3D(SingleColorBase):
+    """A single planar face in 3D space with display properties."""
+
+    type: constr(regex='^DisplayFace3D$') = 'DisplayFace3D'
+
+    geometry: Face3D = Field(
+        ...,
+        description='Face3D for the geometry.'
+    )
+
+    display_mode: DisplayModes = Field(
+        DisplayModes.shaded,
+        description='Text to indicate the display mode (shaded, wireframe, '
+        'etc.). The DisplayModes enumeration contains all acceptable types.'
+    )
+
+
+class DisplayMesh3D(DisplayBaseModel):
+    """A mesh in 3D space with display properties."""
+
+    type: constr(regex='^DisplayMesh3D$') = 'DisplayMesh3D'
+
+    geometry: Mesh3D = Field(
+        ...,
+        description='Mesh3D for the geometry.'
+    )
+
+    colors: List[Color] = Field(
+        ...,
+        description='A list of colors that correspond to either the faces '
+        'of the mesh or the vertices of the mesh. It can also be a single color '
+        'for the entire mesh.'
+    )
```

Comment:
I think we should add a Pydantic validator for this.

1. check the lengths of the inputs and ensure it is either equal to the number of faces or vertices.
2. if there is only a single color, generate a list with the length of the faces.

Reply:
Done!